src/modules/site-v2/base/views/data/data.py

```diff
@@ -57,3 +76,186 @@ def protocols():
     }
     return render_template('data/protocols.html', **params)
 
+
+#
+# Submit Trait
+#
+@data_bp.route('/submit-trait')
+@jwt_required()
+def submit_trait_start():
+  """ Submit Trait start page """
+  return render_template('data/submit-trait-start.html', **{
+    'title': 'Submit Trait',
+    'disable_parent_breadcrumb': True
+  })
+
+#
+# Submit Trait Form
+#
+@data_bp.route('/submit-trait/new-submission', methods=['GET', 'POST'])
+@jwt_required()
+def submit_trait_form():
+  """ Trait Submission Form """
+  form = TraitSubmissionForm()
+  user = get_current_user()
+
+  if hasattr(user, 'username') and not form.username.data:
+    form.username.data = user.username
+
+  # Handle form submission
+  if request.method == 'POST':
```

Comment:
if this function *only* handles `POST` requests, consider doing early-return, eg:
```python
if request.method != 'POST':
  # flash_message('Operation not supported') # or something to that effect. 
  # redirect to a good page. 
```

Reply:
It handles the `GET` request too, which renders the form